crates/goose/src/agents/reference.rs

```diff
@@ -65,7 +68,62 @@ impl Agent for ReferenceAgent {
         let mut messages = messages.to_vec();
         let reply_span = tracing::Span::current();
         let mut capabilities = self.capabilities.lock().await;
-        let tools = capabilities.get_prefixed_tools().await?;
+        let mut tools = capabilities.get_prefixed_tools().await?;
+        // we add in the read_resource tool by default
+        // TODO: make sure there is no collision with another system's tool name
+        let read_resource_tool = Tool::new(
+            "platform__read_resource".to_string(),
+            indoc! {r#"
+                Read a resource from a system.
+
+                Resources allow systems to share data that provide context to LLMs, such as
+                files, database schemas, or application-specific information. This tool searches for the
+                resource URI in the provided system, and reads in the resource content. If no system
+                is provided, the tool will search all systems for the resource.
+
+                The read_resource tool is typically used with a search query (can be before or after). Here are two examples:
+                1. Search for files in Google Drive MCP Server, then call read_resource(gdrive:///<file_id>) to read the Google Drive file.
+                2. You need to gather schema information about a Postgres table before creating a query. So you call read_resource(postgres://<host>/<table>/schema)
+                    to get the schema information for a table and then use to construct your query.
+            "#}.to_string(),
+            json!({
+                "type": "object",
+                "required": ["uri"],
+                "properties": {
+                    "uri": {"type": "string", "description": "Resource URI"},
+                    "system_name": {"type": "string", "description": "Optional system name"}
+                }
+            }),
+        );
+
+        let list_resources_tool = Tool::new(
+            "platform__list_resources".to_string(),
+            indoc! {r#"
+                List resources from a system(s).
+
+                Resources allow systems to share data that provide context to LLMs, such as
+                files, database schemas, or application-specific information. This tool lists resources
+                in the provided system, and returns a list for the user to browse. If no system
+                is provided, the tool will search all systems for the resource.
+
+                The list_resources tool is typically used before a read_resource tool call. Here are two examples:
```

Comment:
nit: i'd remove these extra tips unless we find they are needed tips? not sure if it was tested. i've been noticing these specific examples don't necessarily make it better at using the tools but can confuse it into hallucinating usage that isn't relevant, with frontier models at least